src/GameManager.cpp

```diff
@@ -186,49 +168,40 @@ bool GameManager::createZombie(SDL_Renderer* gRenderer, float x, float y) {
 }
 
 // Deletes zombie from level
-void GameManager::deleteZombie(unsigned int id) {
+void GameManager::deleteZombie(const int32_t id) {
     zombieManager.erase(id);
+
 }
 
-unsigned int GameManager::addObject(Object& newObject) {
-    unsigned int id = 0;
-    if (!objectManager.empty()) {
-        id = objectManager.rbegin()->first + 1;
-    }
+int32_t GameManager::addObject(const Object& newObject) {
+    const int32_t id = generateID();
     objectManager[id] = newObject;
     return id;
 }
 
 // Deletes Object from level
-void GameManager::deleteObject(unsigned int id) {
+void GameManager::deleteObject(const int32_t id) {
     objectManager.erase(id);
 }
 
-unsigned int GameManager::addWeaponDrop(WeaponDrop& newWeaponDrop) {
-    unsigned int id = 0;
-    if (!weaponDropManager.empty()) {
-        id = weaponDropManager.rbegin()->first + 1;
-    }
-    weaponDropManager.insert(std::make_pair(id, newWeaponDrop));
+int32_t GameManager::addWeaponDrop(const WeaponDrop& newWeaponDrop) {
+    const int32_t id = generateID();
+    weaponDropManager.insert({id, newWeaponDrop});
```

Comment:
This method of insertion is inconsistent with above inserts.
Is there are reason for the difference?
Which method should be used for inserting into the various manager maps?